workspaces/liferay-ticket-workspace/client-extensions/liferay-ticket-custom-element/src/pages/AllScreens.tsx

```diff
@@ -0,0 +1,58 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayLayout from '@clayui/layout';
+import {ClayVerticalNav} from '@clayui/nav';
+import {ReactNode} from 'react';
+
+import {useHash} from '../services/useHash';
+import {ScreenType} from '../types';
+import TicketsDashboardApp from './TicketsDashboard';
+import TicketApp from './TicketsOverview';
+
+const routes = [
+	{
+		element: <TicketsDashboardApp screenType={ScreenType.INTEGRATED} />,
+		href: '#dashboard',
+		label: 'Dashboard',
+	},
+	{
+		element: <TicketApp />,
+		href: '#overview',
+		label: 'Tickets Overview',
+	},
+];
+
+const HREF_COMPONENT_MAP: {[key: string]: ReactNode} = {};
```

Comment:
Could the type be `React.FC` instead of `ReactNode` since that's what you've used everywhere else?

Reply:
`useTickets.ts:23 Warning: React has detected a change in the order of Hooks called by AllScreens. This will lead to bugs and errors if not fixed. For more information, read the Rules of Hooks: https://reactjs.org/link/rules-of-hooks

   Previous render            Next render
   ------------------------------------------------------
1. useState                   useState
2. useCallback                useCallback
3. useEffect                  useEffect
4. useCallback                useCallback
5. useContext                 useContext
6. useState                   useState
7. useState                   useRef
   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^`

Reply:
hey @ethib137 ,
I tried many things but I kept getting this error.